src/VisualStudio/IntegrationTest/New.IntegrationTests/InProcess/EditorInProcess.cs

```diff
@@ -891,6 +909,14 @@ public async Task GoToBaseAsync(CancellationToken cancellationToken)
             await TestServices.Editor.WaitForEditorOperationsAsync(cancellationToken);
         }
 
+        public async Task GoToImplementationAsync(CancellationToken cancellationToken)
+        {
+            await TestServices.Shell.ExecuteCommandAsync(WellKnownCommands.Edit.GoToImplementation, cancellationToken);
```

Comment:
Should this method switch to main thread first thing as the existing `GoToBaseAsync` does?

Reply:
Same Q for the GoToDefinitionAsync method above.

Reply:
➡️ No, these methods don't do any work with UI thread affinity that the caller is responsible for